js/adapt-articleView.js

```diff
@@ -1,508 +1,487 @@
-                return $blockContainer.css({"width": "100%"});
-            }
-
-            var $container = this.$el.find(".article-block-slider");
-
-            $blocks.css("width", $container.width()+"px");
-
-            var blockWidth = $($blocks[0]).outerWidth();
-            var totalWidth = $blocks.length * (blockWidth);
-
-            $blockContainer.width(totalWidth + "px");
-
-        },
-
-        _onBlockSliderDeviceChanged: function() {
-            var isEnabled = this._blockSliderIsEnabledOnScreenSizes();
-
-            if (isEnabled) {
-                this.$(".article-block-toolbar, .article-block-bottombar").removeClass("display-none")
-            } else {
-                this.$(".article-block-toolbar, .article-block-bottombar").addClass("display-none");
-            }
-
-            _.delay(function() {
-                $(window).resize();
-            }, 250);
-        },
-
-        _onBlockSliderPageScrollTo: function(selector) {
-            this._disableAnimationOnce = true;
-            _.defer(_.bind(function() {
-                this._disableAnimationOnce = false;
-            }, this));
-
-            if (typeof selector === "object") selector = selector.selector;
-
-            var isEnabled = this._blockSliderIsEnabledOnScreenSizes();
-            if (!isEnabled) {
-                return;
-            }
-
-            if (this.$el.find(selector).length === 0) return;
+      var id = selector.substr(1);
 
-            var id = selector.substr(1);
-
-            var model = Adapt.findById(id);
-            if (!model) return;
+      var model = Adapt.findById(id);
+      if (!model) return;
 
-            var block;
-            if (model.get("_type") == "block") block = model;
-            else block = model.findAncestor("blocks");
-            if (!block) return;
-
-            var children = this.model.getChildren();
-            for (var i = 0, item; item = children.models[i++];) {
-                if (item.get("_id") == block.get("_id")) {
-                    _.defer(_.bind(function() {
-                        this._blockSliderMoveIndex(i-1, false);
-                    }, this));
-                    return;
-                }
-            }
-        },
+      var block;
+      if (model.get("_type") == "block") block = model;
+      else block = model.findAncestor("blocks");
+      if (!block) return;
 
-        _onBlockSliderPageScrolledTo: function() {
-            _.defer(_.bind(function() {
-                this._blockSliderScrollToCurrent(false);
-            }, this));
-        },
-
-        _onBlockSliderRemove: function() {
-            this._blockSliderRemoveEventListeners();
-        },
-
-        _blockSliderRemoveEventListeners: function() {
-            this.$(".component").off("resize", this._blockSliderResizeHeight);
-            this.stopListening(Adapt, "device:changed", this._onBlockSliderDeviceChanged);
+      var children = this.model.getChildren();
```

Comment:
this line and the following `for` loop could be simplified to:
```js
this.model.getChildren().forEach((item, index) => {
  if (item.get('_id') !== block.get('_id')) return;
  _.defer(() => this._blockSliderMoveIndex(index, false));
});
```

Reply:
Just to note that `.forEach()` doesn't allow breaking the loop after finding a match which the current code does on line 464. There's no also other ES6 code in the PR so could be worth leaving the eye candy (and associated framework dependency bump) for another iteration...

Reply:
> Just to note that `.forEach()` doesn't allow breaking the loop after finding a match which the current code does on line 464. 

That's a fair point, how about this instead?
```js
const index = this.model.getChildren().findIndex(model => {
  return model.get('_id') === block.get('_id');
});
if (index !== -1) _.defer(() => this._blockSliderMoveIndex(index, false));
```

> There's no also other ES6 code in the PR so could be worth leaving the eye candy (and associated framework dependency bump) for another iteration...

Also a fair point, though it would only mean bumping it from v5 to v5.5 so not a huge jump... considering this is still yet to be merged and and that FW is already at v5.7 maybe it's not that big an issue?